File: tests/typesafety/typescript/literal_flag.ts
// expect: passes

export function readLog(path: string, mode: "text" | "binary"): number {
  const recordSize = mode === "text" ? 1 : 8;
  return path.length * recordSize;
}

export function tail(): number {
  return readLog("app.log", "binary");
}

File: tests/typesafety/python/literal_flag.py
# expect: passes
# title: The mode parameter accepts only text or binary
# improves: flag_documented
from typing import Literal


def read_log(path: str, mode: Literal["text", "binary"]) -> int:
    record_size = 1 if mode == "text" else 8
    return len(path) * record_size


def tail() -> int:
    return read_log("app.log", "binary")
